presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/ValidateAggregationsWithDefaultValues.java

```diff
@@ -0,0 +1,148 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.sanity;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.metadata.Metadata;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.sql.parser.SqlParser;
+import com.facebook.presto.sql.planner.Symbol;
+import com.facebook.presto.sql.planner.optimizations.ActualProperties;
+import com.facebook.presto.sql.planner.optimizations.PropertyDerivations;
+import com.facebook.presto.sql.planner.optimizations.StreamPropertyDerivations;
+import com.facebook.presto.sql.planner.optimizations.StreamPropertyDerivations.StreamProperties;
+import com.facebook.presto.sql.planner.plan.AggregationNode;
+import com.facebook.presto.sql.planner.plan.ExchangeNode;
+import com.facebook.presto.sql.planner.plan.PlanNode;
+import com.facebook.presto.sql.planner.plan.PlanVisitor;
+import com.facebook.presto.sql.planner.sanity.PlanSanityChecker.Checker;
+
+import java.util.List;
+import java.util.Map;
+
+import static com.facebook.presto.sql.planner.plan.AggregationNode.Step.FINAL;
+import static com.facebook.presto.sql.planner.plan.ExchangeNode.Scope.REMOTE;
+import static com.facebook.presto.sql.planner.plan.ExchangeNode.Type.REPARTITION;
+import static com.google.common.base.Preconditions.checkArgument;
+
+public class ValidateAggregationsWithDefaultValues
+        implements Checker
+{
+    private static final SeenExchanges NO_EXCHANGES_SEEN = new SeenExchanges(false, false);
+
+    private final boolean forceSingleNode;
+
+    public ValidateAggregationsWithDefaultValues(boolean forceSingleNode)
+    {
+        this.forceSingleNode = forceSingleNode;
+    }
+
+    @Override
+    public void validate(PlanNode planNode, Session session, Metadata metadata, SqlParser sqlParser, Map<Symbol, Type> types)
+    {
+        planNode.accept(new Visitor(session, metadata, sqlParser, types), null);
+    }
+
+    private class Visitor
+            extends PlanVisitor<SeenExchanges, Void>
+    {
+        final Session session;
+        final Metadata metadata;
+        final SqlParser parser;
+        final Map<Symbol, Type> types;
+
+        public Visitor(Session session, Metadata metadata, SqlParser parser, Map<Symbol, Type> types)
+        {
+            this.session = session;
+            this.metadata = metadata;
+            this.parser = parser;
+            this.types = types;
+        }
+
+        @Override
+        protected SeenExchanges visitPlan(PlanNode node, Void context)
+        {
+            return aggregatedSeenExchanges(node.getSources());
+        }
+
+        @Override
+        public SeenExchanges visitAggregation(AggregationNode node, Void context)
+        {
+            SeenExchanges seenExchanges = aggregatedSeenExchanges(node.getSources());
+
+            if (!node.getStep().equals(FINAL) || !node.hasEmptyGroupingSet()) {
```

Comment:
Add sth like:
// We are interested only in FINAL aggregations with empty grouping set